pkg/instancetype/compatibility_test.go

```diff
@@ -1,3 +1,4 @@
+//nolint:dupl
```

Comment:
This is disabling and masking any future duplication introduced.

>  dupl is ignored due to false positives in the tests, the suggested
>   duplication being part of a Table with Entries acting on different
>   instance type types with a shared spec.

The linter is just saying that there is duplication of code at large scale here.
IMO this is a valid warning and we should try to fix it directly (extract to functions, name things a bit differently or just structure it differently).

Reply:
The weird thing is that the linter wasn't picking up on the actual duplication here, the spec nested in the very different types/versions of these objects being used. I've removed this anyway and extracted the spec creation into functions.